lib/mysql.js

```diff
@@ -307,17 +310,43 @@ MySQL.prototype.updateOrCreate = function(model, data, options, cb) {
   this._modifyOrCreate(model, data, options, fields, cb);
 };
 
-function dateToMysql(val) {
-  return val.getUTCFullYear() + '-' +
-    fillZeros(val.getUTCMonth() + 1) + '-' +
-    fillZeros(val.getUTCDate()) + ' ' +
-    fillZeros(val.getUTCHours()) + ':' +
-    fillZeros(val.getUTCMinutes()) + ':' +
-    fillZeros(val.getUTCSeconds());
+function dateToMysql(dt, tz) {
+  if (!tz || tz == 'local') {
+    return dt.getFullYear() + '-' +
+      fillZeros(dt.getMonth() + 1) + '-' +
+      fillZeros(dt.getDate()) + ' ' +
+      fillZeros(dt.getHours()) + ':' +
+      fillZeros(dt.getMinutes()) + ':' +
+      fillZeros(dt.getSeconds());
+  } else {
+    tz = convertTimezone(tz);
+
+    if (tz !== false && tz !== 0) {
+      dt.setTime(dt.getTime() + (tz * 60000));
+    }
+    return dt.getUTCFullYear() + '-' +
+      fillZeros(dt.getUTCMonth() + 1) + '-' +
+      fillZeros(dt.getUTCDate()) + ' ' +
+      fillZeros(dt.getUTCHours()) + ':' +
+      fillZeros(dt.getUTCMinutes()) + ':' +
+      fillZeros(dt.getUTCSeconds());
+  }
 
   function fillZeros(v) {
     return v < 10 ? '0' + v : v;
   }
+
+  function convertTimezone(tz) {
+    if (tz === 'Z') {
+      return 0;
+    }
+
+    var m = tz.match(/([\+\-\s])(\d\d):?(\d\d)?/);
+    if (m) {
```

Comment:
One liner for this should be fine: `if (m) return (m[1] == '-' ? -1 : 1) * (parseInt(m[2], 10) + ((m[3] ? parseInt(m[3], 10) : 0) / 60)) * 60;`